File: Leetcode/2785 - Sort Vowels in a String/solution.cpp
bool isVowel(char ch) {
    return (ch == 'a') || (ch == 'e') || (ch == 'i') || (ch == 'o') || (ch == 'u') || (ch == 'A') || (ch == 'E') || (ch == 'I') || (ch == 'O') || (ch == 'U');
}

string sortVowels(string s) {
    vector<char> vowels;
    for (int i = 0; i < s.size(); i++) {
        if (isVowel(s[i])) {
            vowels.push_back(s[i]);
        }
    }

    sort(vowels.begin(), vowels.end());
    for (int i = 0, k = 0; i < s.size(); i++) {
        if (isVowel(s[i])) {
            s[i] = vowels[k++];
        }
    }

    return s;
}

File: Leetcode/2785 - Sort Vowels in a String/solution.py
def sortVowels(self, s: str) -> str:
    def is_vowel(ch: str) -> bool:
        return ch in "aeiouAEIOU"

    vowels: List[str] = [ch for ch in s if is_vowel(ch)]
    vowels.sort()

    result: List[str] = list(s)
    k = 0
    for i, ch in enumerate(result):
        if is_vowel(ch):
            result[i] = vowels[k]
            k += 1

    return "".join(result)